inst/unitTests/runit.sparseConversion.R

```diff
@@ -0,0 +1,593 @@
+    
+    ## p35 (ddiMatrix)
+    d2 <- Diagonal(x = c(10, 1))
+    mtxt <- c("10 0",
+              "0 1")
+    M <- as.matrix(read.table(textConnection(mtxt)))
+    dimnames(M) <- NULL
+    dgc <- as(M, "dgCMatrix")
+    checkEquals(dgc, asSpMat(d2), msg="ddi2dgC")
+    
+    cd2 <- chol(d2)
+    dgc <- as(chol(dgc), "dgCMatrix")
+    checkEquals(dgc, asSpMat(cd2), msg="ddi2dgC")
+    
+    ## p36 (dgCMatrix)
+    m <- Matrix(c(0,0,2:0), 3,5)
+    checkEquals(m, asSpMat(m), msg="dgC2dgC")
+    
+    ## p40 (dgTMatrix)
+    m <- Matrix(0+1:28, nrow = 4)
+    m[-3,c(2,4:5,7)] <- m[ 3, 1:4] <- m[1:3, 6] <- 0
+    mT <- as(m, "dgTMatrix")
+    dgc <- as(m, "dgCMatrix")
+    checkEquals(dgc, asSpMat(mT), msg="dgT2dgC")
+    
+    ## p40 (dgTMatrix)
+    T2 <- new("dgTMatrix",
+              i = as.integer(c(1,1,0,3,3)),
+              j = as.integer(c(2,2,4,0,0)), x=10*1:5, Dim=4:5)
+    dgc <- as(T2, "dgCMatrix")
+    checkEquals(dgc, asSpMat(T2), msg="dgT2dgC")
+    
+    # p42 (ddiMatrix)
+    mtxt <- c("1 0 0",
+              "0 1 0",
+              "0 0 1")
+    M <- as.matrix(read.table(textConnection(mtxt)))
+    dimnames(M) <- NULL
+    dgc <- as(M, "dgCMatrix")
+    checkEquals(dgc, asSpMat(Diagonal(3)), msg="ddi2dgC")
+    
+    mtxt <- c("1000 0   0",
+              "0    100 0",
+              "0    0   10")
+    M <- as.matrix(read.table(textConnection(mtxt)))
+    dimnames(M) <- NULL
+    dgc <- as(M, "dgCMatrix")
+    checkEquals(dgc, asSpMat(Diagonal(x = 10^(3:1))), msg="ddi2dgC")
+    
+    ## p42 (dgTMatrix)
+    M1 <- Matrix(0+0:5, 2,3)
+    M <- kronecker(Diagonal(3), M1)
+    dgc <- as(M, "dgCMatrix")
+    checkEquals(dgc, asSpMat(M), msg="dgT2dgC")
+    
+    ## p42 (ldiMatrix) (To be continued)
+    ## Diagonal(x = (1:4) >= 2)
+    
+    ## p42 (dsCMatrix)
+    S <- crossprod(Matrix(rbinom(60, size=1, prob=0.1), 10,6))
+    dgc <- as(S, "dgCMatrix")
+    checkEquals(dgc, asSpMat(S), msg="dsC2dgC")
+    
+    SI <- S + 10*.symDiagonal(6)
+    dgc <- as(SI, "dgCMatrix")
+    checkEquals(dgc, asSpMat(SI), msg="dgT2dgC")
+    
+    ## p42 (dtCMatrix)
+    I4 <- .sparseDiagonal(4, shape="t")
+    dgc <- as(I4, "dgCMatrix")
+    checkEquals(dgc, asSpMat(I4), msg="dtC2dgC")
+    
+    ## p43 (ddiMatrix)
+    I5 <- Diagonal(5)
+    mtxt <- c("1 0 0 0 0",
+              "0 1 0 0 0",
+              "0 0 1 0 0",
+              "0 0 0 1 0",
+              "0 0 0 0 1")
+    M <- as.matrix(read.table(textConnection(mtxt)))
+    dimnames(M) <- NULL
+    dgc <- as(M, "dgCMatrix")
+    checkEquals(dgc, asSpMat(I5), msg="ddi2dgC")
+    
+    D5 <- Diagonal(x = 10*(1:5))
+    mtxt <- c("10 0 0 0 0",
+              "0 20 0 0 0",
+              "0 0 30 0 0",
+              "0 0 0 40 0",
+              "0 0 0 0 50")
+    M <- as.matrix(read.table(textConnection(mtxt)))
+    dimnames(M) <- NULL
+    dgc <- as(M, "dgCMatrix")
+    checkEquals(dgc, asSpMat(D5), msg="ddi2dgC")
+    
+    # mtxt <- c("0.1 0    0          0     0",
```

Comment:
This test didn't pass. I guess it's because the rounding precision. I leave it for further check.